service/src/test/java/uk/gov/hmcts/reform/fpl/service/LocalAuthorityUserServiceTest.java

```diff
@@ -2,114 +2,136 @@
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
 import uk.gov.hmcts.reform.ccd.client.CaseAccessApi;
-import uk.gov.hmcts.reform.ccd.client.model.UserId;
+import uk.gov.hmcts.reform.ccd.client.CaseUserApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseUser;
 import uk.gov.hmcts.reform.fpl.config.LocalAuthorityUserLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.config.SystemUpdateUserConfiguration;
 import uk.gov.hmcts.reform.fpl.exceptions.NoAssociatedUsersException;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
 
-import static org.assertj.core.api.Assertions.assertThatThrownBy;
-import static org.mockito.ArgumentMatchers.any;
+import java.util.Set;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.ArgumentMatchers.refEq;
 import static org.mockito.BDDMockito.given;
 import static org.mockito.BDDMockito.willThrow;
-import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 
 @ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {JacksonAutoConfiguration.class, SystemUpdateUserConfiguration.class})
 class LocalAuthorityUserServiceTest {
 
     private static final String AUTH_TOKEN = "Bearer token";
     private static final String SERVICE_AUTH_TOKEN = "Bearer service token";
-    private static final String JURISDICTION = "PUBLICLAW";
-    private static final String CASE_TYPE = "CARE_SUPERVISION_EPO";
     private static final String CASE_ID = "1";
-    private static final String CREATOR_USER_ID = "1";
+    private static final String[] USER_IDS = {"1", "2", "3"};
     private static final String LOCAL_AUTHORITY = "example";
+    private static final String INVALID_LOCAL_AUTHORITY = "invalid local authority";
+    private static final Set<String> caseRoles = Set.of("[LASOLICITOR]","[CREATOR]");
 
-    @Mock
+    @MockBean
     private AuthTokenGenerator authTokenGenerator;
-    @Mock
+
+    @MockBean
+    private CaseUserApi caseUserApi;
+
+    @MockBean
     private CaseAccessApi caseAccessApi;
-    @Mock
+
+    @MockBean
     private LocalAuthorityUserLookupConfiguration localAuthorityUserLookupConfiguration;
 
-    @InjectMocks
+    @MockBean
+    private IdamClient client;
+
+    @Value("${system_update.username}")
+    private String username;
+
+    @Value("${system_update.password}")
+    private String password;
```

Comment:
could we use systemupdateconfiguration class for this?

Reply:
done